spec/lang/step/operators.md

```diff
@@ -15,22 +15,36 @@ impl<M: Memory> Machine<M> {
 
 ```rust
 impl<M: Memory> Machine<M> {
-    fn eval_int_un_op(op: IntUnOp, operand: Int) -> Result<Int> {
+    fn eval_int_un_op(op: IntUnOp, operand: Int, int_ty: IntType) -> Result<Int> {
         use IntUnOp::*;
         ret(match op {
             Neg => -operand,
             BitNot => !operand,
+            CountOnes => {
+                let mut ones = Int::ZERO;
+                let mut remaining_bits = operand;
+                for _ in Int::ZERO..int_ty.size.bits() {
```

Comment:
Some comments here would be good.

Please move the actual counting of ones into a helper function.

Would `while remaining_bits != Int::ZERO` also work?

Will this work correctly for negative numbers?